core/src/OpenACC/Kokkos_OpenACC_Instance.cpp

```diff
@@ -0,0 +1,205 @@
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+// FIXME_OPENACC - macro for workaround implementation in UniqueToken
+// constructor. undef'ed at the end
+#define KOKKOS_IMPL_OPENACC_WORKAROUND
+
+#include <Kokkos_Core.hpp>
+
+#include <Kokkos_OpenACC.hpp>
+#include <OpenACC/Kokkos_OpenACC_Instance.hpp>
+#include <OpenACC/Kokkos_OpenACC_Exec.hpp>
+
+#include <vector>
+#include <ostream>
+#include <string>
+
+namespace Kokkos {
+namespace Experimental {
+namespace Impl {
+uint32_t OpenACCInternal::impl_get_instance_id() const noexcept {
+  return m_instance_id;
+}
+
+void OpenACCInternal::fence(openacc_fence_is_static is_static) {
+  fence(
+      "Kokkos::Experimental::Impl::OpenACCInternal::fence: Unnammed Internal "
+      "Fence",
+      is_static);
+}
+void OpenACCInternal::fence(const std::string& name,
+                            openacc_fence_is_static is_static) {
+  if (is_static == openacc_fence_is_static::no) {
+    Kokkos::Tools::Experimental::Impl::profile_fence_event<
+        Kokkos::Experimental::OpenACC>(
+        name,
+        Kokkos::Tools::Experimental::Impl::DirectFenceIDHandle{
+            impl_get_instance_id()},
+        [&]() {
+          //[DEBUG] disabled due to synchronous behaviors of the current
+          // parallel construct implementations. acc_wait_all();
+        });
+  } else {
+    Kokkos::Tools::Experimental::Impl::profile_fence_event<
+        Kokkos::Experimental::OpenACC>(
+        name,
+        Kokkos::Tools::Experimental::SpecialSynchronizationCases::
+            GlobalDeviceSynchronization,
+        [&]() {  //[TODO]: correct device ID
+          acc_wait_all();
```

Comment:
This should be `acc_wait(stream_id)` at some point, right?

Reply:
`//[TODO]: correct device ID` is deleted since it is wrong comment.